src/app/hooks/useViewTracker/index.jsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState, useContext, useRef } from 'react';
+import { useInView } from 'react-intersection-observer';
+// Polyfill IntersectionObsesrver, e.g. for IE11
+import 'intersection-observer';
+
+import { ServiceContext } from '#contexts/ServiceContext';
+import { RequestContext } from '#contexts/RequestContext';
+import { buildATIClickParams } from '#containers/ATIAnalytics/params';
+import { getComponentInfo } from '#app/lib/analyticsUtils/index';
+import { sendEventBeacon } from '#containers/ATIAnalytics/beacon';
+
+const VIEWED_DURATION_MS = 1000;
+
+const useViewTracker = ({ pageData, componentName, actionLabel } = {}) => {
+  let eventTrackingProps;
+  const requestContext = useContext(RequestContext);
+  const serviceContext = useContext(ServiceContext);
+  const timer = useRef(null);
+  const [viewSent, setViewSent] = useState(false);
+  const [ref, inView] = useInView({
+    threshold: 0.5,
+  });
+
+  try {
+    eventTrackingProps = buildATIClickParams(
+      pageData,
+      requestContext,
+      serviceContext,
+    );
+  } catch (error) {
+    eventTrackingProps = null;
+  }
+
+  useEffect(() => {
+    let componentInfo;
+
+    try {
+      componentInfo = getComponentInfo({
+        result: window.location.href,
+        componentName,
+        componentData: {
+          actionLabel,
+          child: 'link', // TODO ask Jon B what this should be
+        },
+      });
+    } catch (error) {
+      componentInfo = null;
+    }
+
+    if (inView && !timer.current) {
+      timer.current = setTimeout(() => {
```

Comment:
Interesting I've never seen `ref`s used like this I thought you used refs to reference DOM elements not store variable data. Feels like state should be being used here instead but very happy to be corrected?

Reply:
refs can be used to hold mutable values. the good thing about using refs over state is that it doesn't trigger a rerender when it updates https://reactjs.org/docs/hooks-reference.html#useref

Reply:
Thanks for that, you are right that's a better API for this use case 👍